src/main/java/net/zhuoweizhang/raspberryjuice/PythonModuleGenerator.java

```diff
@@ -0,0 +1,46 @@
+package net.zhuoweizhang.raspberryjuice;
+
+import org.bukkit.entity.EntityType;
+
+public class PythonModuleGenerator {
```

Comment:
What is the purpose behind the PythonModuleGenerator? 

Generating python code within RaspberryJuice doesnt feel like the right approach. 

Having as a separate utility program seems more appropriate rather than core piece of functionality.

Reply:
The python code needs to be generated from within the plugin because only the plugin knows the relationship between entity names and entity ids on the server. If the server decides to change the ids for a particular entity name, it would be a maintenance nightmare to try to keep the python code in sync any way other than a server plugin. I guess you could create another plugin purely for this purpose but for now it is embedded in RaspberryJuice plugin.

Since I wrote this code I have discovered the RaspberryJam mod also has a spawnEntity method but it uses the names from Minecraft client rather than Minecraft server (which are different for some reason) and I can't see any way to automatically generate this code to be compatible with RaspberryJam, sorry.

Reply:
Ok, I see what you are trying to implement. I think a more appropriate API would be something like:

    world.getEntityTypes()

which would return a list of entity ids and entity names.

By taking this approach you can use it to create the constraints in the python module but leave it open to be used for other things.

Reply:
I will remove the python module generator and implement an api similar to this.

Reply:
So, I have changed Entity to have an additional property `name` and created a new API world.getEntityTypes which returns a list of Entity objects.

Reply:
You can see the changes I have made in the dev branch.

Reply:
OK. I have just perused your changes and they look good. I will test them now. I am surprised you are removing the multiplayer code because I thought it worked fine. I didn't use it much, but one use I had was when trying to get the attention of a room full of children all connected to my minecraft server, I would run my jail script which stopped everyone where they were ;)

import mcpi.minecraft as minecraft
import time

mc = minecraft.Minecraft.create()
listIds = mc.getPlayerEntityIds()
mapOriginalPos = {}
for id in listIds:
    mapOriginalPos[id] = mc.entity.getPos(id)

while True:
    time.sleep(1)
    for id in listIds:
        mc.entity.setPos(id, mapOriginalPos[id])

Reply:
Looks good in tests. I have added a pull request with some changes. Adding names to entity.py hasn't solved the compatibility problem with RaspberryJam mod. RaspberryJam sends entity names not numerical ids as part of api. RaspberryJam uses minecraft client names (eg Ozelot) while we are using minecraft server names (eg Ocelot) so we can't just change api to accept names.